extensions/data-transfer/portability-data-transfer-smugmug/src/main/java/org/datatransferproject/transfer/smugmug/photos/SmugMugPhotosImporter.java

```diff
@@ -116,11 +116,14 @@ String importSinglePhoto(
       PhotoModel inputPhoto,
       SmugMugInterface smugMugInterface)
       throws IOException {
-    String newAlbumUri = idempotentExecutor.getCachedValue(inputPhoto.getAlbumId());
-    checkState(
-        !Strings.isNullOrEmpty(newAlbumUri),
-        "Cached album URI for %s is null",
-        inputPhoto.getAlbumId());
+    try {
+        String newAlbumUri = idempotentExecutor.getCachedValue(inputPhoto.getAlbumId());
+    } catch (IllegalArgumentException) {
+        String newAlbumUri = idempotentExecutor.executeAndSwallowIOExceptions(
```

Comment:
Let's comment. Create the album on demand as all albums are not returned in the initial call.